### ISSUE TITLE
`Style/RedundantRegexpArgument` auto-correction produces syntax error with escaped single quotes

### ISSUE THREAD

**Issue by moznion** [rubocop/rubocop#14711]
The auto-correction for `Style/RedundantRegexpArgument` produces invalid Ruby syntax when the regexp contains escape character(s) and a single quote.

--------

## Steps to reproduce the problem

```ruby
str.gsub!(/\\'/, "'")
```

Then apply RuboCop auto-correction.

## Expected behavior

The auto-correction should produce valid Ruby syntax, such as:

```ruby
str.gsub!('\\\'', "'")
```

## Actual behavior

`Style/RedundantRegexpArgument` rule rewrites it as

```ruby
str.gsub!('\\'', "'")
```

This is a Ruby syntax error.

## RuboCop version

```
$ bundle exec rubocop -V
1.81.7 (using Parser 3.3.10.0, Prism 1.6.0, rubocop-ast 1.48.0, analyzing as Ruby 3.4, running on ruby 3.4.7) [arm64-darwin25]
```
